nmagent/error.go

```diff
@@ -103,3 +105,13 @@ func (e Error) StatusCode() int {
 func (e Error) Unauthorized() bool {
 	return e.Code == http.StatusUnauthorized
 }
+
+// NotFound reports whether the error was produced as a result of the resource
+// not existing.
+func (e Error) NotFound() bool {
+	deleteNetworkPattern := regexp.MustCompile(`/NetworkManagement/joinedVirtualNetworks/[^/]+/api-version/\d+/method/DELETE`)
```

Comment:
do you waint to compile this regexp every time `NotFound()` is called, or should we hoist it in to a higher scope so that it's only compiled once?

Reply:
The regex doesn't need to be compiled every time, moved it to a higher scope.